src/App.js

```diff
@@ -1,19 +1,62 @@
+import { useState, useReducer, useEffect } from "react";
 import NavBar from "./components/NavBar/NavBar";
 import Products from "./components/Products/Products";
 import SideBar from "./components/SideBar/SideBar";
+import { ProductsContext } from "./ProductsContext";
+import { productReducer } from './ProductsReducer';
+import axios from 'axios';
 
 function App() {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [productsPerPage, setProductsPerPage] = useState(16);
+  const [productParams, setProductParams] = useState({
+    _page: currentPage,
+    _limit: productsPerPage,
+  });
+  const [products, setProducts] = useState([]);
+  const [currentLink, setCurrentLink] = useState('');
+
+  const [params, dispatch] = useReducer(productReducer, productParams);
```

Comment:
Lúc này đống này sẽ được sửa lại thành
```suggestion
  const [state, dispatch] = useReducer(productReducer, initialState);
```